dartagnan/src/main/java/com/dat3m/dartagnan/expression/ExpressionFactory.java

```diff
@@ -115,8 +115,8 @@ public Expression makeCast(Expression expression, Type type) {
         return makeCast(expression, type, false);
     }
 
-    public Expression makeConditional(Expression condition, Expression ifTrue, Expression ifFalse) {
-        return new IfExpr(condition, ifTrue, ifFalse);
+    public Expression makeITE(Expression condition, Expression ifTrue, Expression ifFalse) {
```

Comment:
Can we explicitly make the condition a `BoolExp`?

Reply:
No. The problem is that once aggregates are involved, you cannot really have static Java typing anymore:
```
expr = extract(struct(bool, int), 0)
```
Should be a valid expression for the condition, `expr.getType` will be boolean but the static java type will not be `BoolExpr`.